src/navigation/reroute.js

```diff
@@ -0,0 +1,143 @@
+import CustomEvent from 'custom-event';
+import { isStarted } from 'src/start.js';
+import { toLoadPromise } from 'src/child-applications/lifecycles/load.js';
+import { toBootstrapPromise } from 'src/child-applications/lifecycles/bootstrap.js';
+import { toMountPromise} from 'src/child-applications/lifecycles/mount.js';
+import { toUnmountPromise } from 'src/child-applications/lifecycles/unmount.js';
+import { getMountedApps, getAppsToLoad, getAppsToUnmount, getAppsToMount } from 'src/child-applications/child-apps.js';
+import { notSkipped } from 'src/child-applications/child-app.helpers.js';
+import { callCapturedEventListeners } from './navigation-events.js';
+
+let appChangeUnderway = false, peopleWaitingOnAppChange = [];
+
+export function reroute(pendingPromises = [], eventArguments) {
```

Comment:
This is the most important function in all of single-spa. It's what orchestrates the loading, bootstrapping, unmounting, and mounting of child applications.